ledger/store/src/helpers/memory/internal/map.rs

```diff
@@ -100,7 +100,16 @@ impl<
             }
             // Otherwise, insert the key-value pair directly into the map.
             false => {
-                self.map.write().insert(bincode::serialize(&key)?, value);
+                let serialized_key = bincode::serialize(&key)?;
+                let start = std::time::Instant::now();
```

Comment:
Does `Instant` have high precision enough for this? These calls might take less than a millisecond.